### DISCUSSION TITLE
Integrate phyloviz and microreact

### DISCUSSION THREAD

**Discussion by innovate-invent** [phac-nml/irida#1291] [Ideas]
@tom114 @apetkau 
Hello, my name is Nolan Woods, I am working for Dr. Fiona Brinkman.
She would like me to integrate [Phyloviz](http://www.phyloviz.net/) and [Microreact](https://microreact.org/) into IRIDAs visualization suite.
Do you expect any issues with this?

Thanks

**Comment by wwhsiao** [comment]
Instead of Phyloviz, I would suggest GrapeTree for MLST viz.
https://www.ncbi.nlm.nih.gov/pubmed/30049790

Welcome onboard, Nolan!

Cheers,

Will

William Hsiao, PhD
Senior Scientist, BCCDC Public Health Laboratory
Clinical Assistant Professor, Pathology & Laboratory Medicine, UBC
Adjunct Professor, Molecular Biology and Biochemistry, SFU
Rm 4066, 655 West 12th Avenue
Vancouver, BC, V5Z 4R4
Canada
Tel: 604-707-2561
Fax: 604-707-2603

**Comment by innovate-invent** [comment]
@fionabrinkman Will that suit your needs?

**Comment by fionabrinkman** [comment]
Its actually Claire who brought up wanting PhyloViz, so we should check in with her also, though as you know Nolan she also thinks new visualizations need to be implemented! It'd be good to also checkin with Thomas from the NLM and others mentioned to get their input. GrapeTree would indeed also be good. We really want a range of options, and certainly microreact is a nice tool, but note, Will,  for the upcoming meeting of PIs next week we could review these options and come up with a prioritized list. Meanwhile, are you particularly keen on GrapeTree Will for a particular use case, just so I understand the need?

**Comment by innovate-invent** [comment]
ping @cbertell

**Comment by tom114** [comment]
Hi @innovate-invent, thanks for the interest.  A couple opinions here:
* Phyloviz is a great application made by one of our collaborators.  I'd suggest building something like a remote push out to [Phyloviz Online](https://online.phyloviz.net/index) through their API would be an interesting project.  We're still working on functionality to connect to remote applications so this might not be something to start work on immediately (watch issue #99)
* Microreact is kind of in the same boat as Phyloviz.  It's a standalone application that would be a bit odd to directly "integrate" into IRIDA's visualization suite.  We actually currently use the same tree viewer as Microreact in our current visualizations (http://phylocanvas.org/).  I know Microreact also has an API and doing something like a push into their API would be interesting as well (https://microreact.org/api-docs, again relies on #99)
* GrapeTree is one I don't have as much background with, but I know it comes in a stand-alone web server.  It may be a bit more realistic for integration, though I'm not aware of other applications that have built in GrapeTree as an example.

@joshsadam may have other opinions on this and insight on building in some other visualization tools.

**Comment by apetkau** [comment]
Thanks @innovate-invent, integrating these into IRIDA would be awesome :smile: 

I agree with @tom114 that integration through the API is probably the easiest (e.g., pushing data from IRIDA into Phyloviz or Microreact to visualize data) but it depends on how we want people to use the visualization system.

For GrapeTree, it is very fast and can support very large amounts of data. It is incorporated into Enterobase (https://enterobase.readthedocs.io/en/latest/grapetree/grapetree-about.html) and also BigsDB (https://bigsdb.readthedocs.io/en/latest/data_analysis.html#grapetree). I don't know if there is an API which can be used to push data into it but it looks like it is maybe more reasonable for direct integration (at least, it's been done before).

**Comment by joshsadam** [comment]
@apetkau and @tom114 have covered pretty much everything as far as I can see.  I think we need to spend a little bit of time trying to decide what should actually be built into the IRIDA code base versus pushing out to external applications (even if we host these external applications for security reasons).

**Comment by wwhsiao** [comment]
Hi all,

I think my reason for GrapeTree is same as what Aaron said.  Joao is involved in the GrapeTree as well and I think it’s just more scalable then PhyloViz online.  Though I have to admit I have only limited experience with both.

Cheers,

Will

William Hsiao, PhD
Senior Scientist, BCCDC Public Health Laboratory
Clinical Assistant Professor, Pathology & Laboratory Medicine, UBC
Adjunct Professor, Molecular Biology and Biochemistry, SFU
Rm 4066, 655 West 12th Avenue
Vancouver, BC, V5Z 4R4
Canada
Tel: 604-707-2561
Fax: 604-707-2603

**Comment by happykhan** [comment]
You can just pipe your newick & metadata into GrapeTree.  The visual component is just js + html so that just needs to be hosted somewhere on your site & then you point it to the meta and tree. The files can be external; if you can get around CORS.

https://enterobase.readthedocs.io/en/latest/grapetree/grapetree-api.html

**Comment by apetkau** [comment]
Awesome, thanks @happykhan :smile:

**Comment by innovate-invent** [comment]
Thanks everyone for this advice, and sorry for the delay. I am just coming back to this project, along with another visualization that requires integration.

Was there ever a discussion about IRIDA hosted visualizations vs 3rd party visualizations?
Is IRIDA specifically keeping visualizations out of scope for its code?

I also have the option of hosting the visualization in the Galaxy backend, but that would require bypassing IRIDA. This visualization is purely client-side so where it is actually hosted is not critical. 

Is there any documentation related to adding some way for the user to access a visualization for an analysis?

Thanks

**Comment by fionabrinkman** [comment]
There is a new item that came up that might be a good start: Adding in ChewBBACA into IRIDA which is needed to meet a grant milestone. I’ve cc’d also Will who may comment more on this too, based on a recent email thread. 

 https://github.com/B-UMMI/chewBBACA/ <https://github.com/B-UMMI/chewBBACA/>


Best wishes,
Fiona


__

Dr. Fiona S. L. Brinkman, PhD FRSC
Distinguished Professor, 
Molecular Biology and Biochemistry Dept
Associate Professor, School of Computing Science,
and Faculty of Health Sciences
Rm SSB 8166, 8888 University Drive
Simon Fraser University, Burnaby, BC, Canada, V5A 1S6
Phone: 778-782-5646  Fax: 778-782-5583  Office: SSB 7110
http://www.brinkman.mbb.sfu.ca

**Comment by fionabrinkman** [comment]
Sorry, I meant to address this to Nolan!
